tensorboard/plugins/npmi/npmi_plugin.py

```diff
@@ -191,6 +192,34 @@ def values_impl(self, ctx, experiment):
         contents = json.dumps(result)
         return contents
 
+    def embeddings_impl(self, ctx, experiment):
+        mapping = self._data_provider.list_tensors(
+            ctx,
+            experiment_id=experiment,
+            plugin_name=self.plugin_name,
+            run_tag_filter=provider.RunTagFilter(
+                tags=[metadata.EMBEDDINGS_TAG]
+            ),
+        )
+        result = {run: {} for run in mapping}
+        for (run, _) in six.iteritems(mapping):
+            all_embeddings = self._data_provider.read_tensors(
+                ctx,
+                experiment_id=experiment,
+                plugin_name=self.plugin_name,
+                run_tag_filter=provider.RunTagFilter(
+                    runs=[run], tags=[metadata.EMBEDDINGS_TAG]
+                ),
+                downsample=self._downsample_to,
+            )
+            embeddings = all_embeddings.get(run, {}).get(
+                metadata.EMBEDDINGS_TAG, {}
+            )
+            event_data = embeddings[0].numpy.tolist()
+            result[run] = event_data
+        contents = json.dumps(result)
```

Comment:
FYI, you can skip this and our `Respond()` helper will take care of it for you:
https://cs.opensource.google/tensorflow/tensorboard/+/master:tensorboard/backend/http_util.py;l=148;drc=7894225a85ab955aaf998ff4b97bf6fbc911e6ca

Which makes writing tests a little nicer.

Reply:
Thanks